src/server/.tests/ws-routes/controllers/Friends.spec.js

```diff
@@ -0,0 +1,497 @@
+const FriendsProvider = require('../../../ws-routes/controllers/Friends')
+const namespaceInfo = require('../../../protocol/protocol.json').friends
+const config = require('../../../config.json')
+const socketClient = require('socket.io-client')
+const io = require('socket.io')
+const _ = require('lodash')
+const sinon = require('sinon')
+const assert = require('chai').assert
+
+const CLIENT_EVENTS = namespaceInfo.eventTypes
+const SERVER_PORT = config.devPropeties.httpTestPort
+const SERVER_URL = `http://localhost:${SERVER_PORT}`
+const SOCKET_OPTIONS = {
+	transports: ['websocket'],
+	'force new connection': true
+}
+let suite = {}
+
+describe('Friends websocket namespace', () => {
+	beforeEach(() => {
+		suite = {}
+		suite.server = io.listen(SERVER_PORT)
+		suite.DUMMY_USERNAME = 'DUMMY_USERNAME'
+		suite.userMiddlewareMock = sinon.stub().callsFake((socket, next) => {
+			socket.request.user = {username: suite.DUMMY_USERNAME}
+			next()
+		})
+		suite.server.use(suite.userMiddlewareMock)
+		
+		suite.connectionsMock = {
+			usersToConnectionsMap: new Map()
+		} 
+		suite.client = {}
+		suite.userModelMock = {}
+
+	})
+
+	afterEach(done => {
+		suite.server.close(done)
+		if(_.isFunction(suite.client.disconnect)) suite.client.disconnect()
+	})
+
+	describe('#Connection', () => {
+		beforeEach(() => {
+			suite.friendsInstance = new FriendsProvider({
+				UserModel: sinon.stub()
+			})
+		})
+
+		it('should add new element to user-socket map', done => {
+			//given
+			suite.server.on(CLIENT_EVENTS.CONNECTION, socket => {
+				suite.friendsInstance.connection(socket, suite.connectionsMock)
+				then()
+			})
+
+			//when
+			suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+
+			//then
+			function then() {
+				const hasSocketStored = suite.connectionsMock.usersToConnectionsMap.has(suite.DUMMY_USERNAME)
+				assert.isTrue(hasSocketStored)
+				done()
+			}
+		})
+
+	})
+
+	describe('#Friends list', () => {
+		beforeEach(() => {
+			suite.userModelMock = {
+				findOne: sinon.stub()
+			}
+			suite.friendsInstance = new FriendsProvider({
+				UserModel: suite.userModelMock
+			})
+		})
+
+		it('should sent event to client who requested for it', done => {
+			//given
+			const QUERY_FEEDBACK_MOCK = []
+			const queryResultMock = {
+				exec: sinon.stub().resolves(QUERY_FEEDBACK_MOCK)
+			}
+			suite.userModelMock.findOne.returns(queryResultMock)
+
+			suite.server.on(CLIENT_EVENTS.CONNECTION, connection => {
+				connection.on(CLIENT_EVENTS.FRIENDS_LIST, data => {
```

Comment:
What does event `CLIENT_EVENTS.FRIENDS_LIST` mean? Adding to friend list? Removing from friend list?

Reply:
ok